packages/live-status-gateway/src/topics/bucketsTopic.ts

```diff
@@ -0,0 +1,136 @@
+}
+
+export interface BucketStatus {
+	id: string
+	name: string
+	adLibs: BucketAdLibStatus[]
+}
+
+const SHOW_STYLE_BASE_KEYS = ['sourceLayerNamesById', 'outputLayerNamesById'] as const
+type ShowStyle = PickArr<ShowStyleBaseExt, typeof SHOW_STYLE_BASE_KEYS>
+
+export class BucketsTopic extends WebSocketTopicBase implements WebSocketTopic {
+	private _buckets: Bucket[] = []
+	private _adLibActionsByBucket: Record<string, BucketAdLibAction[]> | undefined
+	private _adLibsByBucket: Record<string, BucketAdLib[]> | undefined
+	private _sourceLayersMap: ReadonlyMap<string, string> = new Map()
+	private _outputLayersMap: ReadonlyMap<string, string> = new Map()
+
+	constructor(logger: Logger, handlers: CollectionHandlers) {
+		super(BucketsTopic.name, logger, THROTTLE_PERIOD_MS)
+
+		handlers.bucketsHandler.subscribe(this.onBucketsUpdate)
+		handlers.bucketAdLibActionsHandler.subscribe(this.onBucketAdLibActionsUpdate)
+		handlers.bucketAdLibsHandler.subscribe(this.onBucketAdLibsUpdate)
+		handlers.showStyleBaseHandler.subscribe(this.onShowStyleBaseUpdate)
+	}
+
+	sendStatus(subscribers: Iterable<WebSocket>): void {
+		const buckets: BucketStatus[] = this._buckets.map((bucket) => {
+			const bucketId = unprotectString(bucket._id)
+			const bucketAdLibs = (this._adLibsByBucket?.[bucketId] ?? []).map((adLib) => {
+				const sourceLayerName = this._sourceLayersMap.get(adLib.sourceLayerId)
+				const outputLayerName = this._outputLayersMap.get(adLib.outputLayerId)
+				return literal<BucketAdLibStatus>({
+					id: unprotectString(adLib._id),
+					name: adLib.name,
+					sourceLayer: sourceLayerName ?? 'invalid',
+					outputLayer: outputLayerName ?? 'invalid',
+					actionType: [],
+					tags: adLib.tags,
+					externalId: adLib.externalId,
+					publicData: adLib.publicData,
+				})
+			})
+			const bucketAdLibActions = (this._adLibActionsByBucket?.[bucketId] ?? []).map((action) => {
+				const sourceLayerName = this._sourceLayersMap.get(
+					(action.display as IBlueprintActionManifestDisplayContent).sourceLayerId
+				)
+				const outputLayerName = this._outputLayersMap.get(
+					(action.display as IBlueprintActionManifestDisplayContent).outputLayerId
+				)
+				const triggerModes = action.triggerModes
+					? action.triggerModes.map((t) =>
+							literal<AdLibActionType>({
+								name: t.data,
+								label: interpollateTranslation(t.display.label.key, t.display.label.args),
+							})
+					  )
+					: []
+				return literal<BucketAdLibStatus>({
+					id: unprotectString(action._id),
+					name: interpollateTranslation(action.display.label.key, action.display.label.args),
+					sourceLayer: sourceLayerName ?? 'invalid',
+					outputLayer: outputLayerName ?? 'invalid',
+					actionType: triggerModes,
+					tags: action.display.tags,
+					externalId: action.externalId,
+					publicData: action.publicData,
+				})
+			})
+			return {
+				id: bucketId,
+				name: bucket.name,
+				adLibs: [...bucketAdLibs, ...bucketAdLibActions],
```

Comment:
Should this be sorted according to the `_rank` of the original objects? I feel like it probably should, considering this is returned as an ordered array? As a Sofie user, I would expect the order to be retained (since I can reorder them in the UI). Especially, since if they aren't sorted, any ordering is gone and there's no way of recovering it. I would expect the users to be using reordering for Bucket organization (keeping the "hottest" clip at the begining or whatever else makes sense in one's workflow). I see that `adLibsTopic` keeps sorting, so maybe we could use a similar thing here.

Reply:
I added sorting in the latest commit